src/test/java/armazem/ArmazemTest.java

```diff
@@ -0,0 +1,191 @@
+        armazem.cadastrarIngredienteEmEstoque(ingrediente);
+        IllegalArgumentException ex = assertThrows(IllegalArgumentException.class,
+                () -> armazem.cadastrarIngredienteEmEstoque(ingrediente));
+        assertEquals(ex.getMessage(), INGREDIENTE_JA_CADASTRADO);
+    }
+
+    @Test
+    @DisplayName("[DescadastrarIngrediente]: quando remover Ingrediente, ele nao deveria ser encontrado no estoque")
+    public void descadastrarIngrediente_naoDeveriaEstarNoEstoqueAposSerRemovido() {
+        Ingrediente ingrediente = new Base(TipoBase.IORGUTE);
+        armazem.cadastrarIngredienteEmEstoque(ingrediente);
+        armazem.descadastrarIngredienteEmEstoque(ingrediente);
+
+        IllegalArgumentException ex = assertThrows(IllegalArgumentException.class,
+                () -> armazem.consultarQuantidadeDoIngredienteEmEstoque(ingrediente));
+
+        assertEquals(ex.getMessage(), INGREDIENTE_NAO_ENCONTRADO);
+    }
+
+    @Test
+    @DisplayName("[DescadastrarIngrediente]: quando nao encontrar ingrediente, deveria jogar exception")
+    public void descadastrarIngrediente_exceptionQuandoNaoEncontrarIngrediente() {
+
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+                () -> armazem.descadastrarIngredienteEmEstoque(new Fruta(TipoFruta.BANANA)));
+
+        assertEquals(exception.getMessage(), INGREDIENTE_NAO_ENCONTRADO);
+    }
+
+    @Test
+    @DisplayName("[AdicionarQuantidade]: quando adicionar quantidade no  estoque, quantidade deveria ser atualizada")
+    public void adicionarQuantidade_quantidadeDeveriaSerAtualizada() {
+
+        Ingrediente ingrediente = new Base(TipoBase.IORGUTE);
+        armazem.cadastrarIngredienteEmEstoque(ingrediente);
+
+        armazem.adicionarQuantidadeDoIngredienteEmEstoque(ingrediente, 3);
+        Integer qtd = armazem.consultarQuantidadeDoIngredienteEmEstoque(ingrediente);
+        assertEquals(qtd, 3);
+        armazem.adicionarQuantidadeDoIngredienteEmEstoque(ingrediente, 2);
+        qtd = armazem.consultarQuantidadeDoIngredienteEmEstoque(ingrediente);
+        assertEquals(qtd, 5);
+    }
+
+
+    @Test
+    @DisplayName("[AdicionarQuantidade]: quando adicionar quantidade menor ou igual zero, deveria jogar exception")
+    public void adicionarQuantidade_exceptionQuandoAdicionarQtdMenorOuIgualAZero() {
+        // parametrizado
+        Ingrediente ingrediente = new Base(TipoBase.IORGUTE);
+        armazem.cadastrarIngredienteEmEstoque(ingrediente);
+
+        IllegalArgumentException ex = assertThrows(IllegalArgumentException.class,
+                () -> armazem.adicionarQuantidadeDoIngredienteEmEstoque(ingrediente, 0));
+        IllegalArgumentException ex1 = assertThrows(IllegalArgumentException.class,
+                () -> armazem.adicionarQuantidadeDoIngredienteEmEstoque(ingrediente, -1));
+
+        assertEquals(ex.getMessage(), QUANTIDADE_INVALIDA);
+        assertEquals(ex1.getMessage(), QUANTIDADE_INVALIDA);
+    }
```

Comment:
Neste caso o ideal seria separar os dois testes. Ter um para quando a quantidade for igual a zero e outro para quando for menor que zero. Mas está certinho!!

Reply:
estava pensando em usar aquilo de testes parametrizados depois, mas tenho que estudar um pouco mais essa implementação

Reply:
Boa, acho super válido você tentar mesmo fazer os testes parametrizados, um desafio bem legal para você!!
E em relação a separar os testes, é só para você criar um método que testa quando a quantridade for igual a zero e outro quando a quantidade for menor que zero. Porque por mais que a exceção é a mesma, a lógica é outra. Então o interessante mesmo seria você separar a responsabilidade de cada teste. Você implementou certinho, seria somente quebrar em dois mesmo, separando eles.